source/more_sampling_tools.Rmd

```diff
@@ -209,13 +211,513 @@ message(n_golds / (n_golds + n_silvers))
+
+The code above implements the procedure we might well use if were were
+simulating the problem physically.  We do a trial, and we record the result.
+We do this on a piece of paper if we are doing a physical simulation, and in
+the `second_trunks` {{< var array >}} in code.
+
+Finally we tally up the results.  If we are doing a physical simulation, we go
+back over the all the trial results and counting up the "Gold" and "Silver"
+outcomes.  In code we use the comparisons `== 'Gold'` and `== 'Silver'` to find
+the trials of interest, and then count them up with {{< var sum >}}.
+
+Boolean {{< var array >}}s are a fundamental tool in {{< var lang >}}, and we will use them in nearly all our simulations.
+
+Here is a remind of how those {{< var array >}}s work.
+
+First, let's slice out the first 10 values of the `second_trunks`
+trial-by-trial results tally from the simulation above:
+
+```{python}
+# Get values at positions 0 through 9 (up to, but not including position 10)
+first_10_trunks = second_trunks[:10]
+first_10_trunks
+```
+
+```{r}
+# Get values at positions 1 through 10
+first_10_trunks <- second_trunks[1:10]
+first_10_trunks
+```
+
+Before we started the simulation, we set `second_trunks` to contain 10,000
+strings, where each string was "No gold in first trunk".  In the simulation, we
+check whether there was gold in the first trunk, and, if not, we don't change
+the value in `second_trunk`, and the value remains as "No gold in first trunk".
+
+Only if there was gold in the first trunk, do we go on to check whether the
+second trunk contains silver or gold.  Therefore, we only set a new value in `second_trunks` where there was gold in the first trunk.
+
+Now let us show the effect of running a comparison on `first_10_trunks`:
+
+```{python}
+were_gold = first_10_trunks == 'Gold'
+were_gold
+```
+
+```{r}
+were_gold <- first_10_trunks == 'Gold'
+were_gold
+```
+
+The `== 'Gold'` *comparison* is asking a question.  It is asking that question
+of {{< var an_array >}}, and the {{< var array >}} contains multiple values.
+{{< var np_or_r >}} treats this comparison as asking the question *of each
+element in the {{< var array >}}*.  We get an answer for the question *for each
+element*.  The answer for position [1]{.r}[0]{.python} is {{< var true >}} if
+the element at position [1]{.r}[0]{.python} is equal to `'Gold'` and {{< var
+false >}} otherwise, and so on, for positions [2]{.r}[1]{.python},
+[3]{.r}[2]{.python} and so on.  We started with 10 strings.  After the
+comparison `== 'Gold'` we have 10 Boolean values, where a Boolean value can
+either be {{< var true >}} or {{< var false >}}.
+
+<!---
+We need an illustration here of form:
+
+True        "Gold"
+True        "Gold"
+False       "No gold in first trunk"
+True        "Gold"
+False   =   "Silver"                     == 'Gold'
+...
+
+Any takers?  Good techniques?   Maybe generated Markdown table with transparent boerder for second and fourth column?
```

Comment:
Question: Do you also use html if it is more convenient than markdown?

Reply:
You're right, markdown tables are not very convenient. We tend to generate them, or edit them in editors with support, or display Pandas dataframes (we have some utility functions to make that easier). But I agree that HTML tables would be quite easy to parse, and it looks like Quarto supports that out of the box:

https://quarto.org/docs/authoring/tables.html#html-tables

Reply:
After reading the section, I'm wondering what the goal of the illustration is.
Printing the input data, along with the boolean output, makes the connection quite clear?

Reply:
I personally find pictures rather helpful - there's no new information - just an easier way to grok the information.

I was thinking - we might also be able to generate LaTeX via an `asis` code block.